src/charm.py

```diff
@@ -172,6 +208,35 @@ def _is_configuration_valid(self) -> bool:
             valid = False
         return valid
 
+    # TODO: Use when relation with Or8cr is ready.
+    @staticmethod
+    def _is_valid_domain(domain: str) -> bool:
+        """Validates given domain.
+
+        Args:
+            domain (str): domain.
+
+        Returns:
+            bool: True if domain is valid, False otherwise.
+        """
+        if not domain:
+            return False
+        if re.match(
+            "(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]",  # noqa: W605, E501
+            domain,
+        ):
+            return True
+        return False
+
+    # TODO: Use when relation with Or8cr is ready.
+    @staticmethod
+    def _is_valid_rootca_pem_path(rootca_pem_path: str) -> bool:
```

Comment:
Will we need it when we start using relation?

Reply:
Agree, removed.